contrib/tests/shpproj.sh

```diff
@@ -19,8 +19,19 @@ readonly SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
 
 "${SHPDUMP:-$top_builddir/shpdump}" -precision 8 "test" > "test.out"
 
+is_windows() {
+	[ -n "$WINDIR" ] || uname | grep -q "MINGW"
+}
 
-if result=$(diff --strip-trailing-cr "$SCRIPTDIR/expect.out" "test.out"); then
+run_diff() {
+	if is_windows; then
+		diff --strip-trailing-cr "$SCRIPTDIR/expect.out" "test.out"
```

Comment:
any reason not to apply --strip-trailing-cr unconditionally ? This would avoid the is_windows() which might potentially not catch all situations

Reply:
Do you mean we need to check if diff offers the --strip-trailing-cr option?

Reply:
as far as I can see, both GNU diff (https://man7.org/linux/man-pages/man1/diff.1.html) and BSD diff (https://man.freebsd.org/cgi/man.cgi?diff) have this option, so we hopefully can use it unconditionally

Reply:
According to https://github.com/OSGeo/shapelib/pull/111#issuecomment-2286097239 it is not available on BSD (not sure if OpenBSD or FreeBSD is meant).

Reply:
or maybe run dos2unix on the output to remove CR ?

Reply:
> According to [#111 (comment)](https://github.com/OSGeo/shapelib/pull/111#issuecomment-2286097239) it is not available on BSD (not sure if OpenBSD or FreeBSD is meant).

yes, [OpenBSD's diff](https://man.openbsd.org/diff) doesnt have it